### PR TITLE
Feature/improve redux state creation

### PR DESCRIPTION
**Pull request by cds-amal**
This PR allows DApp developers to configure drizzle with their own middleware, sagas and reducers. One use case would be for DApp devs to listen to contract events. #147

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,9 +35,14 @@ Drizzle is a collection of front-end libraries that make writing dapp frontends
    const drizzle = new Drizzle(options)
    ```
 
-   **Note**: The above assumes you have no existing redux store and generates a new one. To use your existing redux store, see our documentation for [Using an Existing Redux Store](https://www.truffleframework.com/docs/drizzle/using-an-existing-redux-store).
-
-1. Get contract data. Calling the `cacheCall()` function on a contract will execute the desired call and return a corresponding key so the data can be retrieved from the store. When a new block is received, Drizzle will refresh the store automatically _if_ any transactions in the block touched our contract. For more information on how this works, see [How Data Stays Fresh](#how-data-stays-fresh).
+   **Note**: The above assumes you have no existing redux store and generates a new one. In the case you need something a little more sophisticaaated, consult our documentation for [using an Existing Redux Store](https://www.truffleframework.com/docs/drizzle/using-an-existing-redux-store) or [delegating Redux concerns to Drizzle](https://www.truffleframework.com/docs/drizzle/use-drizzles-redux-store)
```
Comment (adrianmcli) [inline README.md]: - [ ] typo: "sophisticaaated"
- [ ] style (be concise): "If you need something more sophisticated..."
- [ ] style (be concise/capitalization consistency): "consult Using an Existing Redux Store or Delegating Redux Concerns to Drizzle"

File: src/contractStateUtils.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+const getAbi = contractEntry =>
+  contractEntry.web3Contract
+    ? contractEntry.web3Contract.options.jsonInterface
+    : contractEntry.abi
+
+export const generateContractInitialState = contractConfig =>
```
Comment (adrianmcli) [inline src/contractStateUtils.js]: Consider refactoring with more intermediate variables and a step-by-step approach for readability/maintainability:

```js
export const generateContractInitialState = contractConfig => {
  const isConstant = x => x.type === "function" && x.constant === true;
  const constants = getAbi(contractConfig).filter(isConstant);
  const objectOfConstants = constants.reduce(
    (acc, x) => ({ ...acc, [x.name]: {} }),
    {}
  );
  return {
    initialized: false,
    synced: false,
    ...objectOfConstants
  };
};
```

**Review by adrianmcli — CHANGES_REQUESTED** [review]
A couple changes here. Do try to proof-read a bit more before asking for a review next time. Otherwise, good work.

File: src/contractStateUtils.js (hunk 2 above)
Comment (cds-amal) [reply src/contractStateUtils.js]: definitely more readable. thanks!

File: README.md (hunk 1 above)
Comment (cds-amal) [reply README.md]: updated, thanks. I'll be more careful with text.

**Comment by adrianmcli** [comment]
lgtm

**Pull request merged**